### PR TITLE
 134249151 Users should be able to chat in group while they are in game

### PR DESCRIPTION
**Pull request by andela-emurphy**
#### What does this PR do?
Add group chat to game any game  session.                
#### Description of Task to be completed?
When a user starts a game, a chat session is automatically created for them. All users in that game session can chat with each other.                
#### How should this be manually tested?                    
- Clone branch.             
- Run 'npm install'.
- Run nodemon.
- Navigate to localhost:3000 on your browser.
- Start a game.
#### Any background context you want to provide?                           
After every gaming session, all history of that chat is deleted

### CODE REVIEW

File: public/css/style.css [hunk 1]
```diff
@@ -1255,3 +1255,89 @@
...
+  margin: 0px 4px;
+  padding-bottom: 10px;
+
+}
+.chat-body .content p {
+  font-size :12px;
+   margin: 0px 4px;
+   word-wrap: break-word;
+}
+.chat .chat-box #chat-form{
```
Comment (adebayo-maborukoje) [inline public/css/style.css]: ### NITPICK
add space in between the element and the brace

File: public/css/style.css [hunk 2]
```diff
@@ -1255,3 +1255,89 @@
...
+  border: 0px;
+  width: 100%;
+  height: 50px;
+  outline: none;
+  color: #545454;
+  word-wrap: break-word;
+}
+.chat-body .left .img, .chat-body .left .content{
```
Comment (adebayo-maborukoje) [inline public/css/style.css]: ###NITPICK

add space in between the element and the brace

File: public/js/controllers/chat.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      $scope.chats = game.cfhChat;
+      $scope.chatWindow = false;
+      $scope.form = {};
+
+      /** checks if chat message is from the user
+       * 
+       * @param{String} username of the user
+       */
+      $scope.isUser = (user) => {
+        return $scope.chats.username === user ? true : false;
```
Comment (adebayo-maborukoje) [inline public/js/controllers/chat.js]: you can just return this `return $scope.chats.username === user` as this in itself is a boolean.

File: public/js/controllers/game.js [hunk 4]
```diff
@@ -80,14 +81,14 @@ angular.module('mean.system')
...
       return game.curQuestion.numAnswers > 1 && 
-        $scope.pickedCards[0] === card.id;
+      $scope.pickedCards[0] === card.id;
     };
 
-    $scope.showSecond = (card) => {
+    $scope.showSecond = function(card) {
       return game.curQuestion.numAnswers > 1 && 
-        $scope.pickedCards[1] === card.id;
+      $scope.pickedCards[1] === card.id;
```
Comment (adebayo-maborukoje) [inline public/js/controllers/game.js]: indent this line as it is a continuation of the previous line. and not a new statement

File: public/js/controllers/game.js [hunk 5]
```diff
@@ -98,8 +99,9 @@ angular.module('mean.system')
       return $index === game.playerIndex;
     };
 
-    $scope.isCustomGame = () => {
-      return !(/^\d+$/).test(game.gameID) && game.state === 'awaiting players';
+    $scope.isCustomGame = function() {
+      return !(/^\d+$/).test(game.gameID) && 
+      game.state === 'awaiting players';
```
Comment (adebayo-maborukoje) [inline public/js/controllers/game.js]: indent this line

File: public/js/controllers/game.js [hunk 6]
```diff
@@ -171,16 +173,17 @@ angular.module('mean.system')
...
-            setTimeout(() => {
-              $scope.link = document.URL;
-              $('#lobby-how-to-play').html('<button class='+
-                '"btn btn-info btn-lg" data-toggle="modal" '+
-                'data-target="#inviteModal">Invite Friends</button>');
+          // Once the game ID is set, update the URL if 
+          //this is a game with friends,where the link is meant to be shared.
+          $location.search({game: game.gameID});
+          if(!$scope.modalShown){
+            setTimeout(function(){
```
Comment (adebayo-maborukoje) [inline public/js/controllers/game.js]: lets use es6 convention here.

File: public/views/chat.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<div class="chat" ng-controller="ChatCtrl">
+  <div class="chat">
+    <div ng-click="showChat()" class="chat-head">
+      <p>Banasko chat!</p>
+      <span> {{ chat.messageCount }}</span>
```
Comment (adebayo-maborukoje) [inline public/views/chat.html]: what is this doing here?

File: public/js/controllers/chat.js (hunk 3 above)
Comment (andela-emurphy) [inline public/js/controllers/chat.js]: done

**Pull request merged**